src/server/generic_family.cc

```diff
@@ -738,13 +738,13 @@ void GenericFamily::Keys(CmdArgList args, ConnectionContext* cntx) {
   string_view pattern(ArgS(args, 1));
   uint64_t cursor = 0;
 
-  StringVec keys;
+  auto output_limit = absl::GetFlag(FLAGS_keys_output_limit);
 
   ScanOpts scan_opts;
   scan_opts.pattern = pattern;
-  scan_opts.limit = 512;
-  auto output_limit = absl::GetFlag(FLAGS_keys_output_limit);
+  scan_opts.limit = std::min(output_limit, 512U);
```

Comment:
I do not think we need `std::min` here. The purpose of the flag is to allow responses greater than the default, not smaller.